src/components/common/Observer.vue

```diff
@@ -0,0 +1,24 @@
+<template>
+  <div class="observer"></div>
+</template>
+
+<script>
+  export default {
+    data: () => ({
+      observer: null
+    }),
+    mounted() {
+      this.observer = new IntersectionObserver(([entry]) => {
+        if (entry && entry.isIntersecting) {
+          this.$emit('murrIntersect')
+        }
+      })
+
+      this.observer.observe(this.$el)
+    }
```

Comment:
I think you still need to add disconnection
```
    destroyed() {
      this.observer.disconnect();
    }
```

Reply:
done